File: src/com/starflask/UltraBlackBloodDeath.java
package com.starflask;

import com.jme3.app.Application;
import com.jme3.app.FlyCamAppState;
import com.jme3.material.Material;
import com.jme3.math.ColorRGBA;
import com.jme3.post.FilterPostProcessor;
import com.jme3.post.filters.BloomFilter;
import com.jme3.scene.Geometry;
import com.jme3.scene.shape.Box;
import com.jme3.system.AppSettings;
import com.jme3.system.JmeContext;
import com.starflask.display.WindowBuilder;
import com.starflask.gameinterface.*;
import com.starflask.states.GameState; 

public class UltraBlackBloodDeath extends MonkeyApplication {

	static WindowBuilder windowBuilder;
	
	public static void main(String args[])
	{
		
		UltraBlackBloodDeath app = new UltraBlackBloodDeath();
		
		//AppSettings settings = new AppSettings(false);
		//settings.setAudioRenderer(AppSettings.JOAL);
		//app.setSettings(settings);  
		
	 //app.start( JmeContext.Type.Headless );
		app.start(JmeContext.Type.Display);
	}

	@Override
	public void simpleInitApp() {
		 
	        
	        this.getStateManager().attach( new  GameState() ) ;
	        
	       
	}
	
	
	
	
		
}

File: src/com/starflask/UltraBlackBloodDeath.scala
package com.starflask

import com.starflask.events.CoreEventQueue
import com.starflask.events.CoreEventPublisher.CoreEvent
import com.starflask.events.CoreEventPublisher._
import com.starflask.states.HeadlessGameState
import com.starflask.states._
import com.jme3.system.AppSettings


object UltraBlackBloodDeath {
  
  def main(args: Array[String]): Unit = {
    var app = new UltraBlackBloodDeath();
    
     app.setShowSettings(false);
		var settings = new AppSettings(true);
		//settings.setAudioRenderer(AppSettings.JOAL);
		
   // settings.put("Width", 1280:java.lang.Integer);
   // settings.put("Height", 720:java.lang.Integer);
		
		  settings.put("Width", 300:java.lang.Integer);
      settings.put("Height", 200:java.lang.Integer);

    settings.put("Title", "Sykophant");
		app.setSettings(settings);  
		
	 //app.start( JmeContext.Type.Headless );
    app.start();
  }
}

class UltraBlackBloodDeath extends MonkeyApplication {
  
  
 // static WindowBuilder windowBuilder;
	
 
	 
	 
	 var coreEventQueue = new CoreEventQueue();

	override def simpleInitApp() {
		 
	        
	     //   this.getStateManager().attach( new  GameState() ) ;
	       var terminalState = this.getStateManager.getState(classOf[TerminalState]); 
	     terminalState.terminalMenu.coreEventPublisher.subscribe { ev => responseToCoreEvent(ev) }
	}
	
	override def simpleUpdate(tpf: Float) {
	  coreEventQueue.feedEvents( (ev) => responseToCoreEvent(ev) )
	}
	
	
	  def responseToCoreEvent(event: CoreEvent) = event match {
	  case h: HostServerEvent => hostGame(h); 
	  case g: JoinServerEvent => joinGame(g); 
	  
	  case _ => println("invalid event")
	    
	}
	
	
	def hostGame(ev: HostServerEvent)
	{
	   this.getStateManager().attach( new  HeadlessGameState() ) ;
	   //collapse the terminal
	   
	}
	
	def joinGame(ev: JoinServerEvent)
	{
	 println("joining");
	   this.getStateManager().attach( new  GameState() ) ;
	}
	
}